crates/forge_permission_resolver/src/permissions_resolver_compass.rs

```diff
@@ -0,0 +1,70 @@
+use serde_yaml;
+use std::collections::HashMap;
+use tracing::warn;
+
+#[derive(Debug)]
+pub struct CompassPermissionResolver {
+    api_scopes_map: HashMap<String, Vec<String>>,
+}
+
+impl CompassPermissionResolver {
+    /// Factory method to create a new `CompassPermissionResolver`
+    pub fn new() -> Self {
+        let permissions = load_compass_api_scopes();
+        CompassPermissionResolver {
+            api_scopes_map: permissions,
+        }
+    }
+
+    /// Retrieves permissions for a given key. Prints a warning if the key is missing.
+    pub fn get(&self, key: &str) -> Option<&Vec<String>> {
+        if let Some(permissions) = self.api_scopes_map.get(key) {
+            Some(permissions)
+        } else {
+            warn!(
+                "Warning: compass API '{}' not found in the api->scopes mapping.",
+                key
+            );
+            None
+        }
+    }
+}
+
+impl Default for CompassPermissionResolver {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+// This is for handling https://developer.atlassian.com/cloud/compass/forge-graphql-toolkit/
+fn load_compass_api_scopes() -> HashMap<String, Vec<String>> {
```

Comment:
```suggestion
fn load_compass_api_scopes() -> HashMap<&'static str, Vec<&'static str>> {
```

`str`s created from `include_str!` are `static`, so you don't need to allocate new ones.